client/cluster-client/src/main/kotlin/uk/ac/ebi/biostd/client/cluster/api/SlurmClusterClient.kt

```diff
@@ -0,0 +1,119 @@
+package uk.ac.ebi.biostd.client.cluster.api
+
+import ebi.ac.uk.base.orFalse
+import ebi.ac.uk.coroutines.waitUntil
+import mu.KotlinLogging
+import uk.ac.ebi.biostd.client.cluster.common.JobSubmitFailException
+import uk.ac.ebi.biostd.client.cluster.common.toSlurmJob
+import uk.ac.ebi.biostd.client.cluster.model.Job
+import uk.ac.ebi.biostd.client.cluster.model.JobSpec
+import java.time.Duration.ofSeconds
+import kotlin.Result.Companion.failure
+import kotlin.Result.Companion.success
+
+private val logger = KotlinLogging.logger {}
+
+class SlurmClusterClient(
+    private val logsPath: String,
+    private val sshMachine: String,
+    private val sshKey: String,
+) : ClusterClient {
+    private val sshClient by lazy { SshClient(sshMachine = sshMachine, sshKey = sshKey) }
+
+    override suspend fun triggerJobAsync(jobSpec: JobSpec): Result<Job> {
+        val parameters = mutableListOf("sbatch --output=$logsPath/%J_OUT --error=$logsPath/%J_IN")
+        parameters.addAll(jobSpec.asParameter())
+        val command = parameters.joinToString(separator = " ")
+        logger.info { "Executing command '$command'" }
+
+        return sshClient.runInSession {
+            val (exitStatus, response) = executeCommand(command)
+            return@runInSession asJobReturn(exitStatus, response)
+        }
+    }
+
+    override suspend fun triggerJobSync(
+        jobSpec: JobSpec,
+        checkJobInterval: Long,
+        maxSecondsDuration: Long,
+    ): Job {
+        val job = triggerJobAsync(jobSpec).getOrThrow()
+        await(job, checkJobInterval, maxSecondsDuration)
+        return job
+    }
+
+    /**
+     * Return the slurm job status. PENDING/RUNNING/COMPLETING/COMPLETED
+     */
+    override suspend fun jobStatus(jobId: String): String {
+        suspend fun CommandRunner.runningJobStatus(): String? {
+            val status = executeCommand("squeue --noheader --format=%T --job $jobId").second
+            if (status.isBlank()) return null else return status
+        }
+
+        suspend fun CommandRunner.historicalJobStatus(): String {
+            val command = "sacct --noheader --format=JobID,State --jobs=$jobId | grep \"^$jobId \" | awk '{print \$2}'"
+            return executeCommand(command).second
+        }
+
+        return sshClient.runInSession {
+            val status = runningJobStatus() ?: historicalJobStatus()
```

Comment:
I think we should handle the case when the given job id is not found

Reply:
if job is not found bash command output is empty strimg so status will be "". By handling do mean trigger an exception in this case or does it look for you that emtpy is correct.